components/brave_new_tab_ui/containers/newTab/settings/braveToday/sources.tsx

```diff
@@ -144,24 +146,53 @@ export default function Sources (props: SourcesProps) {
               <DirectFeedItemMenu key={publisher.publisherId} onRemove={onRemoveDirectFeed.bind(undefined, publisher)} />
             </SettingsRow>
           ))}
-          <Styled.FeedInputLabel>
-            Feed URL
-            <Styled.FeedInput type={'text'} value={feedInputText} onChange={onChangeFeedInput} />
-          </Styled.FeedInputLabel>
-          {(feedInputIsValid === FeedInputValidity.NotValid) &&
-            <Styled.FeedUrlError>Sorry, we couldn't find a feed at that address.</Styled.FeedUrlError>
+          <Styled.AddSourceForm onSubmit={e => { e.preventDefault() }}>
+            <Styled.FeedInputLabel>
+              Feed URL
+              <Styled.FeedInput type={'text'} value={feedInputText} onChange={onChangeFeedInput} />
+            </Styled.FeedInputLabel>
+            {(feedInputIsValid === FeedInputValidity.NotValid) &&
+              <Styled.FeedUrlError>Sorry, we couldn't find a feed at that address.</Styled.FeedUrlError>
+            }
+            {(feedInputIsValid === FeedInputValidity.IsDuplicate) &&
+              <Styled.FeedUrlError>Seems like you already subscribe to that feed.</Styled.FeedUrlError>
+            }
+            <Styled.YourSourcesAction>
+              <Button
+                isPrimary
+                scale='small'
+                type='submit'
+                isDisabled={feedInputIsValid !== FeedInputValidity.Valid}
+                isLoading={feedInputIsValid === FeedInputValidity.Pending}
+                onClick={onSearchForSources}
+              >
+                Add source
+              </Button>
+            </Styled.YourSourcesAction>
+          </Styled.AddSourceForm>
+          {(feedInputIsValid === FeedInputValidity.HasResults) &&
+            <Styled.FeedSearchResults>
+              Multiple feeds were found:
```

Comment:
translation?

Reply:
Translations will be handled in a follow-up since there are others already in this component that require to be moved